operator_manifest/cli.py

```diff
@@ -0,0 +1,286 @@
+# SPDX-License-Identifier: BSD-3-Clause
+import argparse
+import json
+import logging
+import os.path
+import sys
+
+from operator_manifest.operator import ImageName, OperatorManifest
+from operator_manifest.resolver import resolve_image_reference
+
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_OUTPUT_EXTRACT = 'references.json'
+DEFAULT_OUTPUT_REPLACE = 'replacements.json'
+
+CLI_LOGGER_FORMAT = '%(message)s'
+
+
+def main(args=None):
+    logging.basicConfig(level=logging.INFO, format=CLI_LOGGER_FORMAT)
+
+    namespace = _make_parser().parse_args(args)
+    if namespace.command == 'extract':
+        extract_image_references(namespace.manifest_dir, output=namespace.output)
+    elif namespace.command == 'replace':
+        replace_image_references(
+            namespace.manifest_dir, namespace.replacements_file, dry_run=namespace.dry_run
+        )
+    elif namespace.command == 'pin':
+        if namespace.output_replace.fileno() == sys.stdout.fileno():
+            raise ValueError('Cannot use stdout for --output-replace parameter')
+        pin_image_references(
+            namespace.manifest_dir,
+            authfile=namespace.authfile,
+            output_extract=namespace.output_extract,
+            output_replace=namespace.output_replace,
+            dry_run=namespace.dry_run,
+        )
+    else:
+        raise ValueError(f'Unknown sub-parser {namespace.command}')
```

Comment:
This is what happens when you run `operator-manifest` with no arguments, could the error be more user-friendly?

Reply:
Not really. `argparse` doesn't allow it to get this far. If it gets here, it's because we messed up the code. See the `tests/test_cli.py:TestArgumentParsing:test_really_unknown_sub_parser` test for further explanation.

Reply:
I only noticed it because I ran it locally, so something does indeed seem to be messed up :D

Reply:
Oh no. I can't reproduce it:
```
$ operator-manifest yolo
usage: operator-manifest [-h] {extract,replace,pin} ...
operator-manifest: error: argument command: invalid choice: 'yolo' (choose from 'extract', 'replace', 'pin')
```

What do you see?

Reply:
Specifying an unknown command works the same way for me, not specifying one at all is the problem:
```shell
$ operator-manifest           
Traceback (most recent call last):
  File "/home/acmiel/RedHat/OSBS/operator-manifest/venv/bin/operator-manifest", line 8, in <module>
    sys.exit(main())
  File "/home/acmiel/RedHat/OSBS/operator-manifest/venv/lib/python3.9/site-packages/operator_manifest/cli.py", line 42, in main
    raise ValueError(f'Unknown sub-parser {namespace.command}')
ValueError: Unknown sub-parser None
```

Reply:
Nice catch! Fixed it.